api/tests/quest/integration/infrastructure/repositories/profile-reward-repository_test.js

```diff
@@ -0,0 +1,35 @@
+import { PROFILE_REWARDS_TABLE_NAME } from '../../../../../db/migrations/20240820101213_add-profile-rewards-table.js';
+import { giveRewardToTheUser } from '../../../../../src/quest/infrastructure/repositories/profile-reward-repository.js';
+import { REWARD_TYPES } from '../../../../../src/shared/domain/constants.js';
+import { databaseBuilder, expect, knex } from '../../../../test-helper.js';
+
+describe('Quest | Integration | Repository | profile-reward', function () {
+  describe('#giveRewardToTheUser', function () {
+    it('should give a reward to the user', async function () {
+      // given
+      const userId = 1;
+      databaseBuilder.factory.buildUser({
+        id: userId,
+      });
+      databaseBuilder.factory.buildQuest({
+        id: userId,
+        rewardType: REWARD_TYPES.ATTESTATION,
+        rewardId: 2,
+        eligibilityRequirements: { toto: 'tata' },
+        successRequirements: { titi: 'tutu' },
+      });
+      await databaseBuilder.commit();
+
+      // when
+      await giveRewardToTheUser({ userId: userId, rewardId: 2 });
```

Comment:
```suggestion
      await giveRewardToTheUser({ userId, rewardId: 2 });
```

Et chopper l'id directement dans une constante liée au `buildQuest`